sanic_openapi/spec.py

```diff
@@ -1,47 +1,125 @@
+    def schemes(self):
+        return getattr(self.config, "API_SCHEMES", ["http"])
+
+    @property
+    def host(self):
+        return getattr(self.config, "API_HOST", None)
+
+    @property
+    def basePath(self):
+        return getattr(self.config, "API_BASEPATH", None)
 
-        self.basePath = getattr(app.config, "API_BASEPATH", None)
+    # --------------------------------------------------------------- #
+    # Authorization
+    # --------------------------------------------------------------- #
 
-        # --------------------------------------------------------------- #
-        # Authorization
-        # --------------------------------------------------------------- #
+    @property
+    def securityDefinitions(self):
+        return getattr(self.config, "API_SECURITY_DEFINITIONS", None)
 
-        self.securityDefinitions = getattr(app.config, "API_SECURITY_DEFINITIONS", None)
-        self.security = getattr(app.config, "API_SECURITY", None)
+    @property
+    def security(self):
+        return getattr(self.config, "API_SECURITY", None)
 
-        self.definitions = {}
-        self.tags = []
-        self.paths = {}
+    @property
+    def definitions(self):
+        return self._definitions
+
+    @property
+    def tags(self):
+        return self._tags
+
+    @property
+    def paths(self):
+        return self._paths
+
+    def _build_definition(self, definition):
+        for key, schema in chain(
+            {key: getattr(definition, key) for key in dir(definition)}.items(),
+            get_type_hints(definition).items(),
+        ):
+            if key.startswith("_"):
+                continue
+            if isinstance(schema, (list, tuple)):
+                for s in schema:
+                    if get_type_hints(s):
+                        self._build_definition(s)
+
+            elif isinstance(schema, SIMPLE_TYPES):
+                pass
+
+            elif get_type_hints(schema):
+                self._build_definition(schema)
+
+            self._definitions.setdefault(
+                definition.__name__, {"type": "object", "properties": {}}
+            )
+            self._definitions[definition.__name__]["properties"][
+                key
+            ] = serialize_schema(schema)
 
     def add_definitions(self, definitions):
-        self.definitions = definitions
+        if isinstance(definitions, list):
+            for definition in definitions:
+                self._build_definition(definition)
+        elif isinstance(definitions, type):
+            self._build_definition(definitions)
+        elif isinstance(definitions, Object):
+            self._definitions = {**self.definitions, **definitions.definitions}
+        elif isinstance(definitions, SIMPLE_TYPES):
+            if definitions.name:
+                self._definitions[definitions.name] = definitions.serialize()
+        elif isinstance(definitions, dict):
+            self._definitions = {**self.definitions, **definitions}
```

Comment:
When I try the example `cars`, this part will add unexpected item into `definitions`, such as this AUTHORIZATION header:
https://github.com/huge-success/sanic-openapi/blob/67aaf34eca5e339c349ef65bd0392cb8a97f184e/examples/cars/blueprints/car.py#L30

I think we might not need to put every dict objects to the `definitions`. 🤔

Reply:
Yeah, good point. I've removed that and the tests all pass still so that's good